src/sushi.sol

```diff
@@ -161,21 +169,28 @@ contract SushiJoin is CropJoin {
         // Allow caging if any assumptions change
         require(
             wards[msg.sender] == 1 ||
-            masterchef.migrator() != initialMigrator ||
-            masterchef.owner() != address(timelockOwner)
+            masterchef.migrator() != migrator ||
+            masterchef.rewarder(pid) != rewarder ||
+            masterchef.owner() != address(timelock)
         , "SushiJoin/not-authorized");
 
         _cage();
     }
-    function cage(uint256 value, string memory signature, bytes memory data, uint256 eta) external {
+    function cage(uint256 value, string calldata signature, bytes calldata data, uint256 eta) external {
         require(live, "SushiJoin/not-live");
 
         // Verify the queued transaction is targetting one of the dangerous functions on Masterchef
         bytes memory callData;
+        bytes memory argData;
         if (bytes(signature).length == 0) {
             callData = data;
+            argData = new bytes(data.length - 4);
```

Comment:
Aside from an expensive failed tx, what happens if `data.length < 4` here?

Reply:
The failure cases of this function are not super important. We only care about if a valid, dangerous transaction [here](https://github.com/sushiswap/sushiswap/blob/master/contracts/governance/Timelock.sol#L115) somehow fails our `cage()`. If the call data is not of the proper abi form in both the signature prepended or standalone variant then [this line](https://github.com/sushiswap/sushiswap/blob/master/contracts/governance/Timelock.sol#L123) will always fail. This is true for any input where `len(signature) == 0` and `len(data) < 4` as there is no function signature.